### PR TITLE
added tests and updated code so tests pass

### PR DESCRIPTION
**Pull request by matthewhatch**
I added some tests and updated Test-TargetResource, Set-TargetResource, and Get-TargetResource by adding two private functions, set-timezone and get-timezone... it makes it easier to mock.

### CODE REVIEW

**Comment by matthewhatch** [comment]
There is a failing test... I'm looking into why its failing.  Once its fixed I'll update my branch.

File: DSCResources/xTimeZone/xTimeZone.psm1 [hunk 1]
```diff
@@ -13,7 +13,7 @@ function Get-TargetResource
     )
 
     #Get the current TimeZone
-    $CurrentTimeZone = Invoke-Expression "tzutil.exe /g"
```
Comment (vors) [inline DSCResources/xTimeZone/xTimeZone.psm1]: What's the diff here? Is it in line endings?

Comment (matthewhatch) [inline DSCResources/xTimeZone/xTimeZone.psm1]: I updated return value to use the private function Get-TimeZone to make it easier to Mock in tests.

Comment (TravisEz13) [inline DSCResources/xTimeZone/xTimeZone.psm1]: more specifically, why did line 16, and line 40 change?  there must be some sort of white space change, can you please explain?

Comment (matthewhatch) [inline DSCResources/xTimeZone/xTimeZone.psm1]: I can't see any difference in the compare, the version in my AddTests branch does not include any trailing white space.  I may have removed white space if it was there. Other than that the lines are the same.

**Comment by TravisEz13** [comment]
One issue I spotted on review, issue #2.  
I also created an issue to create guidance on how to deal with this in the DscResources Project. [Singleton guidance](https://github.com/PowerShell/DscResources/issues/15) 

Usually issues with keys are considered blocking as fixing the issue will cause a breaking change.  I put some initial thoughts on what I think the guidance should be but let me discuss this with the other reviewers and this was an existing issue in the resource.

File: DSCResources/xTimeZone/xTimeZone.psm1 (hunk 1 above)
Comment (vors) [inline DSCResources/xTimeZone/xTimeZone.psm1]: Perhaps, Windows line ending vs Linux line ending (CRLF vs LF)? You can check both versions with some hex editor.

**Pull request merged**